### ISSUE TITLE
[Bug] Opening Case Book via patient window when disease has not yet been discovered softlocks game

### ISSUE THREAD

**Issue by SebinNyshkim** [CorsixTH/CorsixTH#1799]
#### Describe the issue
During a play session I received an emergency fax about patients with Golf Stones. I had already built the necessary operating theatre, so I was under the impression I could treat the emergency.

However, I noticed that the emergency patients stopped halfway and did not go to the operating theatre. I thought I had made a mistake and wanted to open the case book to look up which room was needed for treatment.

However, this softlocks the game, or rather the `onMouseUp` event. The output in the console indicates that an attempt was made to open an entry in the case book that does not exist (yet). I have to exit the game with Alt+F4.

Either the game is generating emergencies for diseases that I have not yet discovered and therefore should not be getting at all or it's missing logic to direct patients to the correct rooms for treatment regardless of whether the disease has been discovered yet or not.

#### Steps to Reproduce
1. *Load save game attached to this ticket*
2. Wait for Golf Stone emergency to come in
3. Accept emergency
4. Wait for patients to exit helicopter, standing still after entering hospital
5. Click on patient
6. Try to open case book via patient detail window
7. The `onMouseUp` event is now locked and doesn't release any mouse button anymore when pressed (even locking the middle mouse button so the game keeps scrolling forever)

#### Save Game
[Autosave9.zip](https://github.com/CorsixTH/CorsixTH/files/5798505/Autosave9.zip)

#### Expected Behavior
Case Book opening, pre-set to the disease the patient is suffering from

*Alternatively:* emergency should not be occurring at all, if disease not discovered yet

#### System Information
**CorsixTH Version:** 0.64

**Operating System:** Windows 10 20H2

**Theme Hospital Version:** GOG.com

#### gamelog.txt
```
Created game with savegame version 138.
Warning: Trying to remove nonexistent room build callback (function: 000002A0BC0F5300) from humanoid (table: 000002A0BC9A6BE0).
Warning: Trying to remove nonexistent room remove callback (function: 000002A0C40DEC90) from humanoid (table: 000002A0BC9A6BE0).
Warning: Trying to remove nonexistent staff callback (function: 000002A0BC0F2510) from humanoid (table: 000002A0BC9A6BE0).
Savegame version is 138 (v0.64), originally it was 138 (v0.64)
Error in buttonup handler: 
Lua\dialogs/fullscreen\drug_casebook.lua:156: attempt to index a nil value (field '?')
stack traceback:
	Lua\dialogs/fullscreen\drug_casebook.lua:156: in method 'updateIcons'
	Lua\dialogs/fullscreen\drug_casebook.lua:144: in method 'selectDisease'
	Lua\dialogs/fullscreen\drug_casebook.lua:89: in local 'constructor'
	Lua\class.lua:74: in global 'UICasebook'
	Lua\dialogs\patient.lua:326: in local 'callback'
	Lua\window.lua:691: in method 'handleClick'
	Lua\window.lua:1652: in field 'onMouseUp'
	Lua\dialogs\patient.lua:232: in method 'onMouseUp'
	Lua\window.lua:1636: in field 'onMouseUp'
	Lua\ui.lua:902: in function <Lua\ui.lua:889>
	(...tail calls...)
	Lua\app.lua:1012: in function <Lua\app.lua:1007>

```

### Additional Info
Here's a video illustrating the issue. Once I click in the Case Book button on the patient detail window the game locks the mouse click and doesn't release it anymore, forcing me to Alt+F4 out of the game.

Seems like an edge case where a nil reference isn't properly caught when the disease isn't discovered yet and therefore does not appear in the case book of the current level; yet emergencies can occur regardless and won't get treated.

https://user-images.githubusercontent.com/13051568/104245110-f5f67f80-5463-11eb-911d-bb07059f3fb3.mp4

**Comment by lewri** [comment]
This one should be indirectly fixed by #1754

**Comment by mugmuggy** [comment]
Patient stopping part way was a bug fix in 0.64, in this scenario but it looks like there are a few more instances where it can fail.

**Comment by lewri** [comment]
Could probably do with an afterload as a safety net I suppose if an emergency snuck through from an old save (maybe just discover the disease to keep it stable)

**Comment by SebinNyshkim** [comment]
I loaded the same save file under Linux with a Trunk version and the emergency fax does not come in and so the game plays like normal. Seems like the issue was fixed with #1754 wherein patients with undiscovered diseases could come in but not get treated.
